### PR TITLE
digitslist

### PR DESCRIPTION
**Pull request by postvakje**
returns a list of the digits of n in arbitrary base b.

### CODE REVIEW

File: sympy/ntheory/factor_.py [hunk 1]
```diff
@@ -1479,3 +1478,37 @@ def eval(cls, n, k=1):
             else:
                 return Mul(*[(p**(k*(e + 1)) - 1)/(p**k - 1) if k != 0
                            else e + 1 for p, e in factorint(n).items()])
+
+
+def digitslist(n, b=10):
+    """
+    Return a list of the digits of n in base b.
+
+    digitslist(n, 2) is equivalent to [int(d) for d in format(n,'b')]
```
Comment (skirpichev) [inline sympy/ntheory/factor_.py]: Any reason for this new function?

Comment (postvakje) [inline sympy/ntheory/factor_.py]: This was requested in http://osdir.com/ml/sympy/2013-11/msg00042.html and the digits function in gmpy only works for bases between 2 and 62.

**Comment by postvakje** [comment]
Looks like the build failed due to a timeout.  Is there a way to restart the build?

**Comment by clsn** [comment]
Would it be better to have a generator than a list?  I wrote up one some months ago that also handles non-integers (yielding a radix point, etc.; even non-integer bases and "Fibonacci base"), could be modified to be sympy-compatible.

**Comment by smichr** [comment]
:repeat: this will need to be merged with an updated master which has changed and made this unmergable :eyeglasses: See **Synchronization with master sympy/sympy** at [the wiki](https://github.com/sympy/sympy/wiki/Development-workflow)

**Comment by smichr** [comment]
I think it would be better to either 1) raise an error for negative input or 2) always return a tuple giving (sign, digits). But as for the need -- when do you need more than the base-2 digits of a number?

**Comment by postvakje** [comment]
The output mimics `str`, `bin`, `oct` and `hex` which only return the sign when the number is negative.  How about returning  the tuple `('-',list of digits)` when the number is negative and just the list of digits when the number is nonnegative?

As for the use case, there are some sequences in OEIS that require computation in arbitary bases.  Also see the comment above:

> This was requested in http://osdir.com/ml/sympy/2013-11/msg00042.html and the digits function in gmpy only works for bases between 2 and 62.

**Comment by smichr** [comment]
> The output mimics str, bin, oct and hex which only return the sign when the number is negative

Yes, but they all return a string. You are returning a list with a string (sometimes) and the rest, ints.

Maybe if you returned a string like "-bb d1 d2 d3" where bb is the base in which the digits are reported and `di` are the digits it would make sense. But the only way to make a good decision is to think of the use case. How are you envisioning that the result will be used? 

The format I gave can easily be queried to find the number of digits (res.count(' ')) or get the digits (res.split()[1:]). If you don't do it like that then you are forcing the user to keep track of the input to know what to do with the output -- in which case you might as well not store the sign info with the output. And if if one doesn't literally keep track of the input, you have to check the output to see if the first item is a string or not to know how to reconstruct the result -- but you have to keep track of the base (in that case), something that bin and hex do with the b and x that are reported in the format. 

If you want to mimic the python functions then I would suggest that you include sign, base hint and digits and return the result as a string. Separating everything with spaces is what will allow you to do so for any integer base.

**Comment by postvakje** [comment]
I agree with you and will use the format you proposed earlier.

**Comment by smichr** [comment]
You might use the format `(bb, d1, d2, ...)` as in `(-2, 1, 0, 1)` for `digitslist(5, 2)`. Do we need the word "list" in the function name? digits (as in gympy) seems fine to me.

**Comment by smichr** [comment]
> format you proposed earlier

I proposed a couple. Whatever you do, I would recommend including the base somehow, too. (Either recent proposal of mine shows ways to do that.)

File: sympy/ntheory/factor_.py [hunk 2]
```diff
@@ -1545,3 +1545,40 @@ def core(n, t=2):
         for p, e in factorint(n).items():
             y *= p**(e % t)
         return y
+
+
+def digits(n, b=10):
+    """
+    Return a list of the digits of n in base b. First item in list is base hint.
```
Comment (smichr) [inline sympy/ntheory/factor_.py]: First item ... -> The first element in the list is b (or -b if n is negative).

I think you can delete the whole 2nd paragraph as it contains redundancies and information that is available in the Examples.

**Comment by smichr** [comment]
Just a minor docstring edit suggestion which can be made with a commit message containing [skip ci] to skip the travis tests. Then I think this is ready to go.

**Pull request merged**